sdk/servicebus/Azure.Messaging.ServiceBus/src/SessionHandlerOptions.cs

```diff
@@ -0,0 +1,107 @@
+        ///     <see cref="AutoComplete"/> = true
+        ///     <see cref="MessageWaitTimeout"/> = 1 minute
+        ///     <see cref="MaxAutoRenewDuration"/> = 5 minutes
+        /// </summary>
+        /// <param name="exceptionReceivedHandler">A <see cref="Func{T1, TResult}"/> that is invoked during exceptions.
+        /// <see cref="ExceptionReceivedEventArgs"/> contains contextual information regarding the exception.</param>
+        public SessionHandlerOptions(Func<ExceptionReceivedEventArgs, Task> exceptionReceivedHandler)
+        {
+            // These are default values
+            this.AutoComplete = true;
+            this.MaxConcurrentSessions = 2000;
+            this.MessageWaitTimeout = TimeSpan.FromMinutes(1);
+            this.MaxAutoRenewDuration = Constants.ClientPumpRenewLockTimeout;
+            this.ExceptionReceivedHandler = exceptionReceivedHandler ?? throw new ArgumentNullException(nameof(exceptionReceivedHandler));
+        }
+
+        /// <summary>Occurs when an exception is received. Enables you to be notified of any errors encountered by the session pump.
+        /// When errors are received calls will automatically be retried, so this is informational. </summary>
+        public Func<ExceptionReceivedEventArgs, Task> ExceptionReceivedHandler { get; }
+
+        /// <summary>Gets or sets the duration for which the session lock will be renewed automatically.</summary>
+        /// <value>The duration for which the session renew its state.</value>
+        public TimeSpan MaxAutoRenewDuration
+        {
+            get => this._maxAutoRenewDuration;
+
+            set
+            {
+                TimeoutHelper.ThrowIfNegativeArgument(value, nameof(value));
+                this._maxAutoRenewDuration = value;
+            }
+        }
+
+        /// <summary>Gets or sets the time to wait for receiving a message.</summary>
+        /// <value>The time to wait for receiving the message.</value>
+        public TimeSpan MessageWaitTimeout
+        {
+            get => this._messageWaitTimeout;
+
+            set
+            {
+                TimeoutHelper.ThrowIfNegativeArgument(value, nameof(value));
+                this._messageWaitTimeout = value;
+            }
+        }
+
+        /// <summary>Gets or sets the maximum number of existing sessions that the User wants to handle concurrently.</summary>
+        /// <value>The maximum number of sessions that the User wants to handle concurrently.</value>
+        public int MaxConcurrentSessions
+        {
+            get => this._maxConcurrentSessions;
+
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(Resources.MaxConcurrentCallsMustBeGreaterThanZero.FormatForUser(value));
+                }
+
+                this._maxConcurrentSessions = value;
+                this.MaxConcurrentAcceptSessionCalls = Math.Min(value, 2 * Environment.ProcessorCount);
```

Comment:
is that reliable cross-platform in `netstandard2.0`?

Reply:
I believe so - https://docs.microsoft.com/en-us/dotnet/api/system.environment.processorcount?view=netframework-4.8